scripts/apps/archive/directives/Export.js

```diff
@@ -0,0 +1,109 @@
+class LinkFunction {
+    constructor(api, config, multi, notify, storage, session, urls, scope) {
+        this.api = api;
+        this.config = config;
+        this.multi = multi;
+        this.notify = notify;
+        this.storage = storage;
+        this.session = session;
+        this.urls = urls;
+        this.scope = scope;
+
+        this.init();
+    }
+
+    /**
+     * @ngdoc method
+     * @name sdExport#init
+     * @private
+     * @description Initializes the directive with default values for the scope
+     */
+    init() {
+        this.scope.formatters = this.config.previewFormats;
+        this.scope.loading = false;
+        this.scope.selectedFormatter = this.storage.getItem('selectedFormatter') ||
+                                        JSON.stringify(this.scope.formatters[0]);
+        this.scope.validate = false;
+        this.scope.exportFile = this.exportFile.bind(this);
+        this.scope.failures = 0;
+        this.scope.error = false;
+        this.scope.err_msg = null;
+    }
+
+    /**
+     * @ngdoc method
+     * @name sdExport#exportFile
+     * @private
+     * @param {string} formatterString - name of the formatter selected
+     * @param {Boolean} validate - item to be validated for publish action
+     * @description Calls 'export' endpoint with the request and downloads file if export was successful
+     */
+    exportFile(formatterString, validate) {
+        this.scope.loading = true;
+        this.storage.setItem('selectedFormatter', formatterString);
+        let formatter = JSON.parse(formatterString);
+        let itemIdList = this.scope.multi ? _.map(this.multi.getItems(), '_id') : [this.scope.item._id];
+
+        this.api.save('export', {}, {item_ids: itemIdList, format_type: formatter.name, validate: validate})
```

Comment:
I think it would give for a nicer API if you would return this promise, so that callers can tell that the export is completed. Ideally you would have to return the `item` on success in the `.then` call too.
```
@returns {Promise}
```